### ISSUE TITLE
[MongoDB] getMore commands seem to be duplicated with MongoDB 4.x

### ISSUE THREAD

**Issue by Grmiade** [DataDog/dd-trace-js#1797]
**Describe the bug**

`getMore` commands seem to be duplicated with MongoDB 4.x but not with MongoDB 3.x.

<img width="445" alt="Capture d’écran 2022-01-29 à 00 32 24" src="https://user-images.githubusercontent.com/11244106/151635727-da617245-b048-48c1-820a-9b198f84c4a4.png">

<details>
  <summary>Logs with MongoDB 4.x</summary>
  
  ```
  {
      "trace_id": "08ae5c4e99e85bd5",
      "span_id": "2498764495a88de0",
      "parent_id": "62bf4cb88a39f7ce",
      "name": "mongodb.query",
      "resource": "find xxxx xxxx",
      "error": 0,
      "meta": {
        "service": "xxxxx",
        "version": "1.0.0",
        "runtime-id": "d3973eba-df8c-40c1-9817-2ab6cb80298f",
        "span.kind": "client",
        "db.name": "xxxx",
        "mongodb.query": "xxxx",
        "out.host": "xxxx",
        "out.port": "27000"
      },
      "metrics": {
        "_dd.measured": 1,
        "_sample_rate": 1,
        "_sampling_priority_v1": 1
      },
      "start": 1643411606923352800,
      "duration": 11294922,
      "service": "xxxxx-mongodb",
      "type": "mongodb"
    },
    {
      "trace_id": "08ae5c4e99e85bd5",
      "span_id": "4c2602e318e63d83",
      "parent_id": "068ff984154a1f31",
      "name": "mongodb.query",
      "resource": "getMore xxxx",
      "error": 0,
      "meta": {
        "service": "xxxxx",
        "version": "1.0.0",
        "runtime-id": "d3973eba-df8c-40c1-9817-2ab6cb80298f",
        "span.kind": "client",
        "db.name": "xxxx",
        "out.host": "xxxx",
        "out.port": "27000"
      },
      "metrics": {
        "_dd.measured": 1,
        "_sample_rate": 1,
        "_sampling_priority_v1": 1
      },
      "start": 1643411606937235000,
      "duration": 4076172,
      "service": "xxxxx-mongodb",
      "type": "mongodb"
    },
    {
      "trace_id": "08ae5c4e99e85bd5",
      "span_id": "068ff984154a1f31",
      "parent_id": "62bf4cb88a39f7ce",
      "name": "mongodb.query",
      "resource": "getMore xxxx",
      "error": 0,
      "meta": {
        "service": "xxxxx",
        "version": "1.0.0",
        "runtime-id": "d3973eba-df8c-40c1-9817-2ab6cb80298f",
        "span.kind": "client",
        "db.name": "xxxx",
        "out.host": "xxxx",
        "out.port": "27000"
      },
      "metrics": {
        "_dd.measured": 1,
        "_sample_rate": 1,
        "_sampling_priority_v1": 1
      },
      "start": 1643411606936939300,
      "duration": 4399902,
      "service": "xxxxx-mongodb",
      "type": "mongodb"
    },
  ```
</details>

<details>
  <summary>Logs with MongoDB 3.x</summary>

  ```
  {
      "trace_id": "7d2e78aa334c9bbe",
      "span_id": "02a5caf2606a6be3",
      "parent_id": "2956d49e7a8fa5a9",
      "name": "mongodb.query",
      "resource": "find xxxx xxxx",
      "error": 0,
      "meta": {
        "service": "xxxxx",
        "version": "1.0.0",
        "runtime-id": "7626c828-44b0-435a-9042-e1f2c48523be",
        "span.kind": "client",
        "db.name": "xxxx",
        "mongodb.query": "xxxx",
        "out.host": "host.docker.internal"
      },
      "metrics": {
        "_dd.measured": 1,
        "_sample_rate": 1,
        "out.port": 27000,
        "_sampling_priority_v1": 1
      },
      "start": 1643412640721205000,
      "duration": 8768555,
      "service": "xxxxx-mongodb",
      "type": "mongodb"
    },
    {
      "trace_id": "7d2e78aa334c9bbe",
      "span_id": "5e0328183a90f940",
      "parent_id": "2956d49e7a8fa5a9",
      "name": "mongodb.query",
      "resource": "getMore xxxx",
      "error": 0,
      "meta": {
        "service": "xxxxx",
        "version": "1.0.0",
        "runtime-id": "7626c828-44b0-435a-9042-e1f2c48523be",
        "span.kind": "client",
        "db.name": "xxxx",
        "out.host": "host.docker.internal"
      },
      "metrics": {
        "_dd.measured": 1,
        "_sample_rate": 1,
        "out.port": 27000,
        "_sampling_priority_v1": 1
      },
      "start": 1643412640733239600,
      "duration": 11612549,
      "service": "xxxxx-mongodb",
      "type": "mongodb"
    },
  ```
</details>

**Environment**

* **Operation system:** node:17.4.0-alpine
* **Node version:** 17.4.0
* **Tracer version:** 1.7.0

**Comment by bengl** [comment]
Thanks for reporting this, @Grmiade. As far as I can tell, nothing in the code ought to cause two spans to happen. Is it possible that the new library is just producing more queries? If not, is there any way you can whittle this down into a reproduction case?

**Comment by Grmiade** [comment]
@bengl I checked on my side and I found the `command` instrumentation should be enough to instrument the `getMore` operation, since MongoDB calls the `command` method into the `getMore` method. As you can see [here](https://github.com/mongodb/node-mongodb-native/blob/main/src/cmap/connection.ts#L583).

The instrumentation of the `command` AND `getMore` methods is probably the reason why we obtain 2 spans. What do you think?
BTW It seems to me, we have the same problem for `killCursors`.
https://github.com/DataDog/dd-trace-js/blob/9540d22d1d2131d245e0d207e7371645da9d50d7/packages/datadog-plugin-mongodb-core/src/unified.js#L65-L72

**Comment by Grmiade** [comment]
For information, this issue seems to be resolved since v2.4.0.
